### PR TITLE
Refactor eos_cli_config_gen router ospf

### PR DESCRIPTION
**Pull request by carlbuchmann**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Code style update (formatting, renaming)
- [x] Refactoring (no functional changes)
- [ ] Documentation content changes
- [ ] Other (please describe):

## Related Issue(s)

N/A

## Component(s) name

`arista.avd.eos_cli_config_gen`

__Eos Templates__: `router-ospf.j2`

## Proposed changes

- [x] Use `arista.avd.defined` test where possible
- [x] Use `arista.avd.default` filter where possible
- [x] Shorten long EOS CLI
- [ ] Remove default values
- [x] Update Molecule to increase coverage
- [ ] Rename tasks to use collection paths: `ansible.builtin.template`
- [ ] Move to concatenation using `~` instead of `+` to automatically convert to `string`
- [ ] Use `x in []` instead of multiple `if / elif / else` blocks
- [ ] Use `| join` instead of oneline `for` loop in documentation

## Submitted files

- [x] router-ospf.j2

## How to test

Molecule should not introduce change.

## Checklist:

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://www.avd.sh/docs/contributing/) document.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/ansible-avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly
- [x] All new and existing tests passed ([`pre-commit`](https://www.avd.sh/docs/installation/development/#python-virtual-environment), `make linting` and `make sanity-lint`).

### CODE REVIEW

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/router-ospf.j2 [hunk 1]
```diff
@@ -1,54 +1,53 @@
...
-{% endif %}
+{%     endif %}
+{%     if router_ospf.process_ids[process_id].default_information_originate is defined %}
+{%         set default_information_originate_cli = "default-information originate"%}
+{%         if router_ospf.process_ids[process_id].default_information_originate.always is arista.avd.defined(true) %}
+{%             set default_information_originate_cli = default_information_originate_cli ~ " always" %}
+{%         endif %}
+   {{ default_information_originate_cli }}
+{%     endif %}
+{%     if router_ospf.process_ids[process_id].redistribute.static is defined %}
```
Comment (titom73) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/router-ospf.j2:39]: Please use `arista.avd.defined` instead

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/router-ospf.j2 [hunk 2]
```diff
@@ -1,54 +1,53 @@
...
-{%           if router_ospf.process_ids[process_id].redistribute.connected.route_map is defined %}
-   redistribute connected route-map {{ router_ospf.process_ids[process_id].redistribute.connected.route_map }}
-{%           else %}
-   redistribute connected
-{%           endif %}
-{%         endif %}
-{%     endfor %}
-{% endif %}
+{%     endif %}
+{%     if router_ospf.process_ids[process_id].default_information_originate is defined %}
```
Comment (titom73) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/router-ospf.j2:32]: Please use `arista.avd.defined` instead

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/router-ospf.j2 [hunk 3]
```diff
@@ -1,54 +1,53 @@
...
+   {{ default_information_originate_cli }}
+{%     endif %}
+{%     if router_ospf.process_ids[process_id].redistribute.static is defined %}
+{%         set redistribute_static_cli = "redistribute static" %}
+{%         if router_ospf.process_ids[process_id].redistribute.static.route_map is arista.avd.defined %}
+{%             set redistribute_static_cli = redistribute_static_cli ~ " route-map " ~  router_ospf.process_ids[process_id].redistribute.static.route_map %}
+{%         endif %}
+   {{ redistribute_static_cli}}
+{%     endif %}
+{%     if router_ospf.process_ids[process_id].redistribute.connected is defined %}
```
Comment (titom73) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/router-ospf.j2:46]: Please use `arista.avd.defined` instead

**Review by titom73 — CHANGES_REQUESTED** [review]
Please use `arista.avd.defined` instead of builtin `defined` filter

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/router-ospf.j2 (hunk 3 above)
Comment (carlbuchmann) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/router-ospf.j2:46]: We can't at this time as with the current data model we allow for an empty dictionary key "redistribute.connected:" to generate: `redistribute connected`

See test case: ansible_collections/arista/avd/molecule/eos_cli_config_gen/inventory/host_vars/router-ospf.yml

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/router-ospf.j2 (hunk 2 above)
Comment (carlbuchmann) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/router-ospf.j2:32]: We can't at this time as with the current data model we allow for an empty dictionary key "default_information_originate:" to generate: `default-information originate`

See test case: ansible_collections/arista/avd/molecule/eos_cli_config_gen/inventory/host_vars/router-ospf.yml

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/router-ospf.j2 (hunk 1 above)
Comment (carlbuchmann) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/router-ospf.j2:39]: We can't at this time as with the current data model we allow for an empty dictionary key "redistribute.static:" to generate: `redistribute static`

See test case: ansible_collections/arista/avd/molecule/eos_cli_config_gen/inventory/host_vars/router-ospf.yml

Comment (titom73) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/router-ospf.j2:39]: Got it.

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/router-ospf.j2 (hunk 2 above)
Comment (titom73) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/router-ospf.j2:32]: Got it.

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/router-ospf.j2 (hunk 3 above)
Comment (titom73) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/router-ospf.j2:46]: Got it.

**Review by titom73 — APPROVED** [review]

**Pull request merged**